packages/studio-web/src/app/shared/wc-styling/wc-styling.component.html

```diff
@@ -0,0 +1,254 @@
+          >Write</mat-button-toggle
+        >
+        <mat-button-toggle i18n="Lable for style File toggle" value="upload"
+          >File</mat-button-toggle
+        >
+      </mat-button-toggle-group>
+    </div>
+    <div class="p-0 order-0 order-md-2">
+      <button
+        mat-icon-button
+        (click)="collapsed = !collapsed"
+        data-test-id="toggle-css-box"
+      >
+        <mat-icon *ngIf="collapsed">keyboard_arrow_down</mat-icon
+        ><mat-icon *ngIf="!collapsed">keyboard_arrow_up</mat-icon>
+      </button>
+    </div>
+  </div>
+  <div *ngIf="$inputType === 'upload'">
+    <div class="row">
+      <div class="mt-4 mb-3">
+        <!-- <div class="container"> -->
+        <label
+          i18n="Instructions for uploading style"
+          for="updateStyle"
+          class="form-label"
+          >Select a style sheet file (.css)</label
+        >
+        <input
+          #styleInputElement
+          (change)="onFileSelected($event)"
+          class="form-control"
+          name="text"
+          type="file"
+          id="updateStyle"
+          accept=".css"
+        />
+        <!-- </div> -->
+      </div>
+    </div>
+  </div>
+  <div *ngIf="$inputType === 'edit'">
+    <div class="d-flex flex-column flex-md-row justify-content-between p-0">
+      <span class="p-0"
+        ><h2 i18n="Title for style writing box" class="subtitle">
+          Write or paste your style sheet here
+        </h2></span
+      >
+      <span class="p-0">
+        <button mat-button (click)="updateStyle()">
+          <mat-icon class="mat-icon-lg">sync</mat-icon>
+          Apply
+        </button></span
+      >
+      <span class="p-0">
+        <button
+          i18n="Style download button"
+          mat-button
+          [disabled]="!($styleText | async)"
+          (click)="downloadStyle()"
+          aria-label="Style download button"
+          class="pr-0"
+        >
+          <mat-icon class="mat-icon-lg">save</mat-icon> Save a copy
+        </button>
+      </span>
+    </div>
+    <div class="row">
+      <div class="mt-4 mb-3">
+        <!-- <div class="container"> -->
+        <label
+          i18n="Instructions for uploading font"
+          for="defaultFont"
+          class="form-label"
+          >Optional: use a custom font (.woff2)</label
+        >
+        <input
+          #fontInputElement
+          (change)="onFontSelected($event)"
+          class="form-control"
+          name="text"
+          type="file"
+          id="defaultFont"
+          accept=".woff2,.woff,.ttf"
+        />
+        <!-- </div> -->
+      </div>
+    </div>
+    <div class="row">
+      <mat-form-field class="col-12 p-0 b-0">
+        <textarea
+          [ngModel]="$styleText | async"
+          (ngModelChange)="$styleText.next($event)"
+          id="styleInput"
+          class="border rounded b-0 p-0 bg-light"
+          matInput
+          placeholder=':root,
```

Comment:
Hum, it's nice that the version of this in the help panel has syntax highlighting, but I wish we didn't repeat it twice like this. That's going to be annoying to maintain whenever we make changes. Any chance we can factor the two out, either with automated syntax highlighting there, or with automated stripping of the syntax highlighting here?

And the test suite has a third copy, too. If you had it factored out, the test suite might be able to use it from the same place too...

Reply:
I think we could shorten the placeholder, not sure how we can dynamically style the code in the helper